sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsDeviceMemoryStore.scala

```diff
@@ -40,23 +40,64 @@ class RapidsDeviceMemoryStore(catalog: RapidsBufferCatalog = RapidsBufferCatalog
    * @param id buffer ID to associate with this buffer
    * @param table cudf table based from the contiguous buffer
    * @param contigBuffer device memory buffer backing the table
+   * @param tableMeta metadata describing the buffer layout
    * @param initialSpillPriority starting spill priority value for the buffer
    */
   def addTable(
       id: RapidsBufferId,
       table: Table,
       contigBuffer: DeviceMemoryBuffer,
+      tableMeta: TableMeta,
       initialSpillPriority: Long): Unit = {
-    val buffer = uncompressedBufferFromTable(id, table, contigBuffer, initialSpillPriority)
+    val buffer = new RapidsDeviceMemoryBuffer(
+      id,
+      contigBuffer.getLength,
+      tableMeta,
+      Some(table),
+      contigBuffer,
+      initialSpillPriority)
+    try {
+      logDebug(s"Adding table for: [id=$id, size=${buffer.size}, " +
+          s"meta_id=${buffer.meta.bufferMeta.id}, meta_size=${buffer.meta.bufferMeta.size}]")
+      addBuffer(buffer)
+    } catch {
+      case t: Throwable =>
+        buffer.free()
+        throw t
+    }
+  }
+
+  /**
+   * Adds a contiguous table to the device storage. This does NOT take ownership of the
+   * contiguous table, so it is the responsibility of the caller to close it. The refcount of the
+   * underlying device buffer will be incremented so the contiguous table can be closed before
+   * this buffer is destroyed.
+   * @param id buffer ID to associate with this buffer
+   * @param contigTable contiguous table to track in storage
+   * @param initialSpillPriority starting spill priority value for the buffer
+   */
+  def addContiguousTable(
+      id: RapidsBufferId,
+      contigTable: ContiguousTable,
+      initialSpillPriority: Long): Unit = {
+    val contigBuffer = contigTable.getBuffer
+    val size = contigBuffer.getLength
+    val meta = MetaUtils.buildTableMeta(id.tableId, contigTable)
+    contigBuffer.incRefCount()
+    val buffer = new RapidsDeviceMemoryBuffer(
+      id,
+      size,
+      meta,
+      None,
```

Comment:
why should this be `None` again? I am just trying to follow that.

Reply:
We don't want to manifest the table, just store the underlying table buffer.  The main change here is to stop assuming it's cheap to get the `Table` instance from a `ContiguousSplit` instance.